src/utils/modelValidationHelper.js

```diff
@@ -1,9 +1,14 @@
-import { isStringNullOrWhitespace } from "./propertyValidator";
+import { isStringNullOrWhitespace, isString } from "./propertyValidator";
 import { validationErrorTypes } from "../constants";
 import { isEmpty } from "lodash";
 
 export const validationRules = {
-	[validationErrorTypes.fieldIsRequired]: value => (isStringNullOrWhitespace(value) ? false : true),
+	[validationErrorTypes.fieldIsRequired]: value => {
+		if (value === null || value === undefined) return false;
+		if (isString(value) && isStringNullOrWhitespace(value)) return false;
```

Comment:
I'm suggesting it make sense to add that isString check inside isStringNullOrWhitespace function. Since that function logically expecting string and if the value passed is not string then probably we would like to return true.

If you think it doesn't make sense feel free to ignore that suggestion.